[18.0][MIG] contract_price_revision

AlvaroRM11 opened the pull request:
@BinhexTeam
Depends on: https://github.com/OCA/contract/pull/1136

cubells committed:
[ADD] contract_price_revision: new module to update contract prices easily

pedrobaeza committed:
[IMP] contract_price_revision: Improve a bit performance

oca-travis committed:
[UPD] Update contract_price_revision.pot

carlosdauden committed:
[MIG] contract_price_revision: Migration to 12.0

astirpe committed:
[FIX] contract_price_revision: Skip revising lines that are not valid anymore
cherry-pick db5c3d647225dde2b20214932a060aa279554b07

sbejaoui committed:
[FIX] - fix development_status

rousseldenis committed:
[IMP] contract_price_revision: black, isort, prettier

rousseldenis committed:
[13.0][MIG] contract_price_revision

rousseldenis committed:
[13.0][IMP] contract_price_revision: Add fixed price revision

rousseldenis committed:
[13.0][FIX] contract_price_revision: Recurring next date is not well computed

On new line creation when applying wizard, the stored values
are not recomputed, so recomputing them

rousseldenis committed:
[13.0][FIX] contract_price_revision: Copy() does the job

rousseldenis committed:
[FIX] contract_price_revision: Use digits string instead of decimal precision

rousseldenis committed:
[IMP] contract_price_revision: Improve readability by adding a computed field 'price_can_be_revised'

rousseldenis committed:
[IMP] contract_price_revision: Improve compatibility with contract base module

Values for new line should come from _prepare_value_for_plan_successor function
adding the new price.

rousseldenis committed:
[IMP] contract_price_revision: Allows to restrict price revision on some lines

As if recurrence management is per line, some lines should be excluded from price revision.

carolinafernandez-tecnativa committed:
[IMP] contract_price_revision: pre-commit stuff

carolinafernandez-tecnativa committed:
[MIG] contract_price_revision: Migration to 15.0

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (23 of 23 strings)

Translation: contract-15.0/contract-15.0-contract_price_revision
Translate-URL: https://translation.odoo-community.org/projects/contract-15-0/contract-15-0-contract_price_revision/es/

juanjosesegui-tecnativa committed:
[IMP] contract_price_revision: pre-commit auto fixes

juanjosesegui-tecnativa committed:
[MIG] contract_price_revision: Migration to 17.0

Commit:
[UPD] Update contract_price_revision.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-17.0/contract-17.0-contract_price_revision
Translate-URL: https://translation.odoo-community.org/projects/contract-17-0/contract-17-0-contract_price_revision/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (22 of 22 strings)

Translation: contract-17.0/contract-17.0-contract_price_revision
Translate-URL: https://translation.odoo-community.org/projects/contract-17-0/contract-17-0-contract_price_revision/it/

christian-ramos-tecnativa commented:
Thanks for your contribution @AlvRM2002 , remember to check the [migration guidelines](contract_price_revision) and do it step by step. Missing the pre-commit stuff right now.

rousseldenis commented:
/ocabot migration contract_price_revision

rrebollo commented:
@AlvRM2002 I can do a code review after you get tests in green.

AdamBark commented:
> @AlvRM2002 I can do a code review after you get tests in green.

What is required to get the tests green? It appears this PR is stuck waiting for the contract module to be updated to v18.

rousseldenis commented:
> > @AlvRM2002 I can do a code review after you get tests in green.
> 
> What is required to get the tests green? It appears this PR is stuck waiting for the contract module to be updated to v18.

@AlvRM2002 @AdamBark In order to test this in runboat and make tests green, you need to follow this : https://github.com/OCA/maintainer-tools/wiki/Use-temporary-reference%28s%29-to-another-pull-request%28s%29

AlvaroRM11 committed:
[MIG] contract_price_revision: Migration to 18.0

[FIX] set active_ids to list

[FIX] contract_price_revision: precommit run

[FIX] contract_price_revision: Change dependency

AlvaroRM11 commented:
> > @AlvRM2002 I can do a code review after you get tests in green.
> 
> What is required to get the tests green? It appears this PR is stuck waiting for the contract module to be updated to v18.

@AdamBark you can now try it with the runbot, thanks for your interest!

acsonefho commented on contract_price_revision/tests/test_contract_price_revision.py:
```diff
@@ -0,0 +1,90 @@
+        )
+
+    def execute_wizard(self):
+        self.wizard.with_context(active_ids=self.contract.id).action_apply()
```
`active_ids` should be a `list`.

Ideally, the model should be specified too:
`active_ids = self.contract.ids, active_model=self.contract._name`

acsonefho reviewed (changes requested):
GJ :muscle:

rousseldenis reviewed (approved):
LGTM

rousseldenis commented:
@AlvaroRM11 Thanks for this.

Could you put depending PR's link in this description in order to easy review? Thanks

AlvaroRM11 commented:
> @AlvaroRM11 Thanks for this.
> 
> Could you put depending PR's link in this description in order to easy review? Thanks

Done!

sbejaoui commented on contract_price_revision/__manifest__.py:
```diff
@@ -0,0 +1,23 @@
+    "author": "ACSONE SA/NV, Tecnativa, Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "website": "https://github.com/OCA/contract",
+    "depends": ["contract"],
```
```suggestion
    "depends": ["contract_line_successor"],
```

rousseldenis commented:
@AlvaroRM11 To get green tests, you should include contract_line_successor PR in `test-requirements.txt` too

sbejaoui commented:
can you rebase please

marcelsavegnago reviewed (approved):
LGTM

victoralmau commented:
Can you review the CI and correct the errors in tests?

victoralmau commented:
Superseed by https://github.com/OCA/contract/pull/1290
